scripts/CMSRunAnalysis.py

```diff
@@ -286,92 +286,67 @@ def handleException(exitAcronym, exitCode, exitMsg):
 def parseArgs():
     parser = PassThroughOptionParser()
     parser.add_option('--jobId', dest='jobId', type='string')
-    parser.add_option('--json', dest='jsonArgFile', type='string')
-    parser.add_option('--userSandbox', dest='userSandbox', type='string')
-    parser.add_option('--inputFileList', dest='inputFileList', type='string')
-    parser.add_option('--jobNumber', dest='jobNumber', type='string')
-    parser.add_option('--cmsswVersion', dest='cmsswVersion', type='string')
-    parser.add_option('--scramArch', dest='scramArch', type='string')
-    parser.add_option('--runAndLumis', dest='runAndLumis', type='string', default=None)
-    parser.add_option('--lheInputFiles', dest='lheInputFiles', type='string', default='False')
-    parser.add_option('--firstEvent', dest='firstEvent', type='string', default=0)
-    parser.add_option('--firstLumi', dest='firstLumi', type='string', default=None)
-    parser.add_option('--lastEvent', dest='lastEvent', type='string', default=-1)
-    parser.add_option('--firstRun', dest='firstRun', type='string', default=None)
-    parser.add_option('--seeding', dest='seeding', type='string', default=None)
-    parser.add_option('--userFiles', dest='userFiles', type='string')
-    parser.add_option('--oneEventMode', dest='oneEventMode', default=0)
-    parser.add_option('--scriptExe', dest='scriptExe', type='string')
-    parser.add_option('--scriptArgs', dest='scriptArgs', type='string')
-    parser.add_option('--eventsPerLumi', dest='eventsPerLumi', type='string', default=None)
-    parser.add_option('--maxRuntime', dest='maxRuntime', type='string', default=None)
 
     (opts, _) = parser.parse_args(sys.argv[1:])
 
-    # sanitize input options
-    # Note about options: default value is None for all, but if, like it usually happens,
-    # CRAB code calls this with inputs like --seeding=None
-    # the option variable is set to the string 'None', not to the python type None
-    # let's make life easier by replacing 'None' with None.
-    for name, value in vars(opts).items():
-        if value == 'None':
-            setattr(opts, name, None)
+    jobId = None if opts.jobId is 'None' else opts.jobId
```

Comment:
I think it would be better to ensure that jobId is present. And CRAB code will not call it with `None` string anymore. Indeed after removing the comments in lines 311-318, if you keep this line it would be very obscure what it is about !